### DISCUSSION TITLE
do you know rocketcontroller din modules ?

### DISCUSSION THREAD

**Discussion by ouinouin** [arendst/Tasmota#16394] [Q&A]
https://rocketcontroller.com/

looks well made with Din modular format , I ll by some and try to give feedback but to my mind its the next big thing  to replace the sonoff 4CH and all those kind of modules sinc it can go directly on the electrical board and moreover seems to support 4 AC input which is nice.

**Comment by sfromis** [comment]
Nice with the firmware being Tasmota from the start.

**Comment by ouinouin** [comment]
I bought one unit with the AC inputs +4 relays and one unit with the raw GPIO + 4 relays available 
i ll make feedback and try to publish template on templates.blakadder.com if not already done in the meantime.

**Reply by barbudor** [reply]
I wonder how the extension is working
It is connected through I2C so I would expect PCF/MCP port extander
All good for the relays
But the diagram shows sensors such as temperature, light or motion sensor that Tasmota cannot manage from an port expander
May be they are using an ADC for that ?

In the DIY section they are reselling the RoboDyn ESP32R4

**Reply by sfromis** [reply]
Since there is I2C, it would be quite feasible to have an ADC chip on the bus, and use analog sensors. Basic digital sensors like a PIR can of course also be seen as analog signals, just with only two levels available.

However, since their "current release" app update says "Support any I2C devices with IC PCF8574", they might have done some trickery there....

**Reply by sfromis** [reply]
Their "firmware configuration" tab on https://rocketcontroller.com/product/astra-basic-controller-4-relays-16a-4-ac-input-ac-power-led-br4a4al/ has 4 gpios for inputs, 4 for relays, one for wifi led, one for buzzer, one for RF, double tx/rx for two serial ports with CS on the 2nd, 4 for SPI, and I2C with an interrupt pin alongside. One model adds gpio 14/15 without specified purpose.

**Reply by barbudor** [reply]
Serial2 CS doesn't really mean something for Tasmota...
I2C INT neither, although some drivers may need an interrupt pin

**Reply by sfromis** [reply]
In principle, using INT from PCF8574 could be used to enhance that driver. Not sure what CS on serial even means, could be CTS but that is meant to be used with also having RTS. Or maybe just a pin available somewhere on the inside....

**Reply by barbudor** [reply]
I've order a controller and an extension

**Reply by sfromis** [reply]
More generally, if they made trickery for extra support in Tasmota, and they distribute the binary, the GPL licence requires them to also make source available. Did not find that, but I may not have looked hard enough. Or I just don't know what happens :stuck_out_tongue_closed_eyes:

**Comment by Jason2866** [comment]
A Chinese Hardware Manufacturer who takes the chance to pay/invest nothing for cloud services and software development and support.

**Comment by ouinouin** [comment]
on this link they host updates for the esp32r4 controller : 
https://rocketcontroller.com/esp32r4-controller-tasmota-app-update-april-2022/
so would also to be checked what trickery are they doing there  (i bought one).
and if we see modifications they did to be published, we can try to contact them.
after all pieces of chinese manufacturer code are generally not a big thing, so if they dont want to finish with bad advertisement on things like hackaday or other techies blogs, they should better play the game.

**Reply by Jason2866** [reply]
Nothing special PCF8574 https://tasmota.github.io/docs/PCF8574/#outputs is standard.

**Reply by sfromis** [reply]
Where it gets interesting is when the expansion box supports sensors, and they have words like "Support any I2C devices with IC PCF8574", that could be read as supporting I2C sensors via PCF8574. While I may be reading chinglish too literally, it looks like something is not standard. Could be their language use, but I'd not make that a firm conclusion. Their sensor hookups does not look like simply I2C plugs, but those sensors have to be connected somehow.

**Reply by barbudor** [reply]
I would read that it supports connecting any extension board based on PCF port extender even if they are not theirs
I don't read that as supporting adding sensors through a PCF

And don't forget that this is related to the RobotDyn ESP32R4, not the ASTRA range

**Reply by sfromis** [reply]
With the [Astra controller](https://rocketcontroller.com/product/astra-basic-controller-4-relays-16a-4-ac-input-ac-power-led-br4a4al/) (which is what I've been looking at) specifically based on Tasmota firmware, the expansion boxes having connections for sensors which does not *appear* to be I2C terminals, the [displayed connection](https://rocketcontroller.com/product/astra-expansion-module-4-relays-16a-4-ac-input-ac-power-er4a4a/) between controller and expansion boxes being only I2C, there needs to be some way of getting sensor data to the controller box. How?

**Reply by barbudor** [reply]
I thought we discussed that earlier
The expansion is likely to include both a PCF for the relays and likely an ADS
The bottom connector being specifically dedicated on sensor (analog?) and not for switches like the connector at the same place on the controller
Neverthess, all wondering will get answers after a proper teardown when we receive our units

**Comment by barbudor** [comment]
@ouinouin Have you received a confirmation of shipping ?
Order is still processing for me and visa have been used.

**Reply by ouinouin** [reply]
nope , i purchased through aliexpress with many other things , for the moment its one of the few that are not told as dispatched

**Reply by barbudor** [reply]
Through Ali? I didn't found the ASTRA on Ali
Could you please share the link?

**Reply by ouinouin** [reply]
@barbudor  here the link https://aliexpress.com/item/1005004464526242.html

its on the robodyn shop

still no shipment for now. even for the r4 box.

**Reply by Jason2866** [reply]
Good pictures in review (Aliexpress). There is nothing else inside other than a power supply, driver transistors for relays and optocouplers for inputs. The only magic thing is a dip switch. Btw the PCB traces to the relays are not capable for 16A or even 10A.

**Reply by barbudor** [reply]
Yes, nothing new under the sun
Still there are not so many din rail integration available on the market, especially with esp32

The AC tracks can be reinforced by soldering stronger rigid wire but of course the question remains the true support for the relay.
Still OK for lights but I won't put my water boiler on that for sure

**Reply by sfromis** [reply]
The usually reasonable strategy is to take "chinese amp rating" and divide by two before considering what to use the thing for. Or maybe halve it again with those unreinforced PCB traces.

**Reply by solidstatenz** [reply]
Have just got the ESP32R4 and x8 relay expander, have been able to get all relays working via  I.P address. 
Have not been able to access the expansion relays through home assistant , the first 4 relays are showing in the "MQTT" integration but the expansion relays are not showing? I have set the correct GPIOS for I2C otherwise it wouldn't be working through the I.P address. have also set up the PCF8574 ( all channels set as output).  It feels like the expander is not sending mqtt to home assistant . Anyone have any ideas please .

**Reply by barbudor** [reply]
@solidstatenz 
Are you using HaTasmota plugin and `SetOption19 0` ? (not the old and obsolete `SetOption19 1`)
If yes, please set `Weblog 4`, restart tasmota and copy/paste here the 2 lines with `tasmota/discovery`

**Reply by solidstatenz** [reply]
Not using HaTasmota plug in , didnt know about it.

**Reply by barbudor** [reply]
If you are using `Setoption19 1` you'd better move to HaTasmota and `Setoption19 0`
Due to lack of developpers interested in HA the old integration method is deprecated

**Reply by solidstatenz** [reply]
thanks for all your help. Im not a python ninja so will have to do some study on how to use it, doesn't seem to be any documentation or tutorials on HaTasmota.

**Comment by barbudor** [comment]
I've received my units yesterday.
Cheap plastic, closed by clips, doesn't give an impression of a pro device
I Will play with them this week end

**Reply by barbudor** [reply]
I have started the controller. 
The RJ cable for the expander was not included but apprently they are sending one.

It came with Tasmota 11.1.0.2

The template is:
```
{"NAME":"ASTRA R4A4","GPIO":[1,1,1,1,576,1,1,1,480,1,1,1,3232,3200,1,1,0,640,608,1,0,224,225,1152,0,0,0,0,227,226,160,161,162,0,0,163],"FLAG":0,"BASE":1}
```

The features in Tasmota are: [decode-status_astra-a4r4_original.txt](https://github.com/arendst/Tasmota/files/9641279/decode-status_astra-a4r4_original.txt)

Very weird : GPIO27 comes defined as RFrecv. As I don't have the RF option, I changed it to User in the Template page.
Then I started to get strange IRReceived messages
```
18:08:58.945 RSL: RESULT = {"Time":"2022-09-25T18:08:58","IrReceived":{"Protocol":"UNKNOWN","Bits":50,"Hash":"0xF9059BD0"}}
18:08:59.499 RSL: RESULT = {"Time":"2022-09-25T18:08:59","IrReceived":{"Protocol":"UNKNOWN","Bits":50,"Hash":"0x3D028BF1"}}
```
and strangely, `gpio 255` listed `"GPIO27":{"1088":"IRrecv"}` while it's not in the template
Very strange

Now reflashed with latest 12.1.1.2 with a custom binary (hardcoded SSID, MQTT and template)

**Reply by nitinuae** [reply]
Hello
I have it too - But since I installed it - i cannot open it again to check.
The only thing which works is the 4 relays (I have 4 relay and 8 input version). On their website, in instructions, Esphome has more details of GPIO.

By default, the temp shown on the webpage is 100+ but it shouldnt be (We are normal people and use C). I tried using option of F/C but it still shows 100+C.

I was able to flash 12.x without any loss of functionality. So, hardcode/custom compile is not required as template is retained acroos flashes.

**Comment by nitinuae** [comment]
Adding-
Mine had 
{"NAME":"ASTRA R4A4","GPIO":[1,1,1,1,576,1,1,1,480,1,1,1,3232,3200,1,1,0,640,608,1,0,224,225,1152,0,0,0,0,227,226,160,161,162,0,0,163],"FLAG":0,"BASE":1}

**Reply by barbudor** [reply]
same as mine 😄

**Reply by nitinuae** [reply]
Hello
Were you able to experiment on this?

**Reply by barbudor** [reply]
On what ?
I have reflashed with my own build of tasmota32
I have received the RJ cable to conect the extender to the controller yesterday so this week-end I will play with both
Still I waiting for some answers from Rocketcontroller about the dip switches

I already received the information that - as expected - the dipswitch on the extender is to select the I2C address (only swiches 1 to 3 are used, switch 4 is not used) - see below

I undersand that the 8-dipswitches on the controller are to switch the GPIO between Serial and SPI but haven't yet received the details

>2. DIP switch need for select I2C address of module. If you want to connect many expansion modules to controller, for each one module need dedicate address.
>Follow this table:
>0 its -
>1 its +
>![image](https://user-images.githubusercontent.com/2996042/193030326-63b82b02-000a-48e0-aeca-ef3f8b115d74.png)
>
>3. After selected address, pls connect expansion module to controller with cable.
>4. Start controller (switch on or reset button).
>5. Find in interface menu PCF8574
>6. For IO0-3 select output, For IO4-7 select input. Save.
>7. Power of Expansion module. Connection all relays output. (Dont connect input)
>8. Power ON expansion module.  Connect all inputs.

**Comment by nitinuae** [comment]
Hello - I meant the pinout. (as you wrote you will experiment with it)
I presume the inputs are PFC (so it will require special build).
I was looking for pinout assignments of the RJ11/12 connector they have used as RS/SPI or I2C. My idea is to use it as additional inputs (I have a requirement to read around 14 dry contacts - so 8 is not cutting it)

**Reply by barbudor** [reply]
The pinout of the connectors are on the Web site.
Go to the page of your controller and open the last item "Interface socket"

Now I have more details about the RS/SPI connector
There is a dip switch inside the box
When all 8 dipswitches up (edit: ON), the connector is in SPI mode
```
      RJ pin     SPI Mode
       1         3V3
       2         GND
       3         SPI MOSI (GPIO23)
       4         SPI MISO (GPIO19)
       5         SPI CLK  (GPIO18)
       6         SPI CS0  (GPIO5)
```
As far as I can see, the RJ is connected straight to the GPIO pins. So up to you to change the definition from SPI to SWITCH and use it as 4 inputs.

The RS mode requires the 8 dipswitch to be down (edit: OFF). In that case you would need a RS module that plugs on the pair of very small black connectors left of the dipswicthes. The top half part (6 pins, labelled UART2) is connected to the ESP's GPIOs and the bottom half part (4 pins, labelled RS) is connected to the RJ connector (through the dipswitch ?)
The goal of this is to host an serial adaptation module like a MAX232 to have true RS232, or may be a MAX485 to have a RS485/Modbus interface. They told me the RS modules are not yet ready to sell.
Still I have big wondering on how they can properly swich between those 2 usages without conflicts by using dip switches that are SPST. It would be better to have [SPDT](https://learn.sparkfun.com/tutorials/button-and-switch-basics/poles-and-throws-open-and-closed)

I will try to understand better the schematic
May be also I will ask them if they are kind enough to share it

**Reply by barbudor** [reply]
I'v made up a schematics of all I/Os for internal and external connectors of the R4D4 controller
It should soon be published as an update on the template page : https://templates.blakadder.com/rocket_R4A4.html

To use SPI mode, you should push all DIPswitches to the ON position (up)
To use RS mode, DIPswitches should be pushed down to the OFF position
RS mode is intended to be used with a RS Adaptation module (for example RS232 or RS485/Modbus) but Rocketcontrolled hasn't finalized them. Without RS module, the RS mode is useless.
If you want to use serial on the RS/SPI without adaptation but directly connected to GPIO, leave the dipswitch on SPI mode but change the GPIOs as below:
- GPIO 16 => None
- GPIO 17 => None
- GPIO 5 => None
- GPIO 18 => None
- GPIO 19 => Serial TX
- GPIO 23 => Serial RX


![Rocketcontroller ASTRA R4D4](https://user-images.githubusercontent.com/2996042/193358682-dfcaa95a-dedb-412e-8c7f-fc594852ae26.png)

**Comment by nitinuae** [comment]
Little bit of confusion - In i2c, there is no need of voltage pin?
In RS/SPI mode, I dont think DIP switch is connected to 3.3v. This 3.3v should be available to pin1 regardless of the DIP setting (same with pin2.)

**Reply by barbudor** [reply]
1) The I2C is designed to be connected to the Expander which has it's own power-supply. So only the I2C signals and the common GND are required.
They didn't had custom sensors in mind, like connecting a BME280. But I understand it could make sense. For that kind of usage, you would have to add the 3V3 connection yourself. That will be fine for 1 or 2 sensors that are low power.
However if you plan to make your own Relay Expander, use a separated power supply as the internal power supply of the controller won't be able to cope with driving additional relays.

2) I have probed all signal and I can insure you that this is how it works. It has some logic.
In SPI mode, 3V3 is provided of RJ pin 1 through the dip siwtch. RS module should not be installed.
In RS mode, the signals to RJ are connected to the internal optional RS module.
Now, no one will prevent you to leave the dipswitch for 3V3 on it's ON position if you want to have 3V3 on that pin.
But understand that "RS mode" means "go through internal RS module"
If you want just basic ESP Serial, leave it in SPI Mode and configure the GPIO as I suggested.

**Reply by nitinuae** [reply]
Its clear now.
I just want to have binary sensor - so dry contacts - Will use SPI mode connections then.
3.3v will be the common - and dry contacts will be read through balance of pins.

As I am still waiting for 6P6C wire (its uncomon here), can you explain how I2C works? Is it high always? (as I can use common 6p4C wire)
If I connect dry contacts to ground, and read each contact, the contact will show 0v (as its shorted to ground). Can this be done?
IO21 = switch
IO22 = switch
IO2 = switch 
in template definition. 
And when dry contact is closed, all the IO is shorted to ground. 
Will it work (As sking as I have never read the ESP secs or IO float voltage)

**Reply by sfromis** [reply]
I2C is bidirectional communication on SDA, with clock on SCL. These pins are pulled up (commonly every breakout board has pullup resistors), active low. Needs GND too, obviously, and power (very often 3.3V, sometimes 5V). With short wires, nearly anything goes. I2C is designed as an internal device bus, on or near the main PCB, **not** longer wires. About 1 meter is already very long, but may work, a lot of the time....

**Reply by nitinuae** [reply]
Dusting my old knowledge - so if I define 
IO21 = switch
IO22 = switch
IO2 = switch

and pull these down to ground through a switch, it will be detected by ESP?(ie, everytine the switch is flicked) 

This way - I can use ready 6p4c wires as I wont be needing the outer 2 wires. Yes - the wires are around 50-60cm long

**Reply by barbudor** [reply]
Correct
The GPIOs are there on the connector
Up to you to redefine the GPIOs as Switch instead of I2C to use them as input

**Reply by nitinuae** [reply]
Thank you for clarifying.

For the designated pins (8 GPIO exposed by rocjetcontroller )

Since you are at it - I saw that the IOs (the specified IO pins) are either pulled up or high in ESP32(and here I am confused again).

So, to remove one wire(and the urden of moving 3.3v as well as GND), I can ground the GPIO to get dry contact status? This way, I run one ground wire, use the designated 8 GPIO exposed AND use the GPIO exposed by RJ12 connector (the SPI connector)?
Will it work (since GPIO pins in ESP32 are pulled internally by ESP32)

**Reply by barbudor** [reply]
If you set the GPIO as `Switch`, Tasmota will enable the ESP32's internal pull up
You can just have to ground the pin to change the state of the Switch. Depending on how you set the SwitchMode, that would mean going from OFF to ON or from OFF to ON or even other options

Be sure that the other circuits are fully isolated from the controller.
If they are using relay outputs to ground the GPIO, that would be fine.

**Comment by nitinuae** [comment]
Hello
Unconnected to above queries (I was able to solve with the help above - thank you), my Astra didnot come with the green "phoenix type connector".
All the phoenix types I have are 3.8mm. This one seems to use some other size. After measuring - it might be 4.2mm or 4.22mm pitch. Does that type of connector exist or it proprietory.

(PS - I have asked question via aliexpress - but the seller isnt communicating- may be I had put dispute and it got me in my favor)

**Reply by barbudor** [reply]
Both my controller and my expander where delivered with the connector.
It's only 5 pins on mine but I can measure around 15.3 mm acros the 5 pins which gives ~3.825mm
So 3.8 seems to be the right value

Now considering that the window for that connector is around 40mm and that there are 12 pins, ~3.825 seems also a correct value

I think the model is 15EDG and it is given for 3.1mm 
Those would probably fit : https://www.aliexpress.com/item/4000660083322.html

**Comment by nitinuae** [comment]
I had the ones lying around from kincony. They are 3.81mm. They dont fit (Rocket one is bit spaced out)

**Comment by ntwb** [comment]
Also somewhat unrelated, but to note there are relatively new esp32r4 available that differs from the current device at https://templates.blakadder.com/robotdyn_ESP32R4.html

The image attached to this PR is v1 or v2,not sure, black PCB https://github.com/blakadder/templates/pull/1199

The new v3 has a green PCB https://www.aliexpress.com/item/1005003153036762.html

My guess is the new devices are from approx August this year, as I got the v1/2 black PCB version in July, and now considering the getting a v3 for the Wrover 16mb flash / 8mb psram 🤤 

p.s. Thanks for all the above info, insightful into the what goes on behind the scenes, cheers 👍🏼

Edit: Also unsure if any of the GPIO pins have changed between version etc

**Comment by dagi0627** [comment]


**Reply by sfromis** [reply]
You send nothing that way. Creating bytes with a number argument allocates that size, but empty. If you want a string argument, it has to be wrapped in `'` or `"`.

**Comment by ortegafernando** [comment]
Hi, does anybody know and email or how to contact with Rocketcontroller's people? I am trying to contact them but ... not luck. Thanks

**Comment by nitinuae** [comment]
The guy was quite responsive on aliexpress message center (I bought from aliexpress)


-------- Original message --------
From: ortegafernando ***@***.***>
Date: 29/11/2022 13:07 (GMT+04:00)
To: arendst/Tasmota ***@***.***>
Cc: nitinuae ***@***.***>, Comment ***@***.***>
Subject: Re: [arendst/Tasmota] do you know rocketcontroller din modules ? (Discussion #16394)


Hi, does anybody know and email or how to contact with Rocketcontroller's people? I am trying to contact them but ... not luck. Thanks

**Comment by jeroenst** [comment]
Is the i2c expansion module like ER4A8A supported by the current version of Tasmota? And how can I configure these?

**Reply by barbudor** [reply]
You need to self compile with PCF8574 options
The driver has been fixed 2..3 days ago for proper initialization so you must use the development branch

**Comment by Th0maz** [comment]
Did anyone put the RocketController Astra into their fuse box successfully?
Mine appears to be a bit too high to fit through the front cover.
All other din rail components are about 45 millimeters in height, while the front cover has a 46 millimeter slot.
The RocketController however is 47 millimeters high :(

![2023-05-17 23_40_47-IMG_20230517_200017](https://github.com/arendst/Tasmota/assets/12065628/bb676db7-414c-4d46-b65f-05ba5679dd94)

**Reply by Visioneer3** [reply]
Yes, they are to high.

**Comment by jeroenst** [comment]
Not in the main fuse box but no problem with it in the housing I used which is a small waterproof fusebox. 

![20230120_151542](https://github.com/arendst/Tasmota/assets/2833940/401a7f96-525d-45c7-8e22-502403329454)

**Reply by orange-seed** [reply]
hi @jeroenst , how do find/ communicate the expansion module and the basic module? i have connected the i2c port on both side of the modules and it seems it doesn't detect the expansion module in the tasmota page..

**Reply by barbudor** [reply]
Do you have the PCF8574 in your build?
Is i2cdriver2 enabled?
What is the result of i2cscan?

**Reply by orange-seed** [reply]
the result of i2cscan is; RESULT = {"I2CScan":"Device(s) found at 0x38"}

**Reply by barbudor** [reply]
That is answering only 1 question out of 3

Beside 0x38 is not a supported PCF8574 address (0x39 to 0x3F)
See https://tasmota.github.io/docs/I2CDEVICES/#supported-i2c-devices
0x38 has be reserved for other devices that do not have the flexibility to change their i2c address

Read the PCF8574 specific documentation https://tasmota.github.io/docs/PCF8574/#report-bugs-and-suggest-features

**Comment by just-grizzle** [comment]
Anyone got an idea if and how one could "easily" add ethernet on this device? I mean by just adding an off the shelf module.
It's the only missing part holding me back from purchase.

**Reply by barbudor** [reply]
I don't see that as an easy task
The ESP32 is on the module below
It could be possible to hack into that a module with Ethernet such as [WT32-ETH01](https://www.aliexpress.com/item/1005004971916518.html) but that would probably be ugly and you would have ethernet connector on the side.

Better look into Sonoff SPM


![2022-09-25 18 13 01](https://github.com/arendst/Tasmota/assets/2996042/512836fc-892d-4b17-9452-d0181f230721)

**Reply by just-grizzle** [reply]
Sonoff SPM is way more expensive whereas I don't need power monitoring. Thx.

**Reply by njh** [reply]
Also Sonoff SPM doesn't have any wall switch inputs.

I don't understand why hardly any of the Sonoff products have switch inputs.
Only the Sonoff Mini?

**Comment by nitinuae** [comment]
Check kincony also.
Its much better build, but DC only.

This Robo module has IO which is directly connected to ESP
________________________________

**Reply by just-grizzle** [reply]
@barbudor @nitinuae Thank you both for your answers. You convinced me not to try fiddling around. 

I''ll check out Kincony. Meanwhile I found Olimex may have something that fits my needs (4 relays, RJ45, RS-485). Just not sure if I could this all working with Tasmota. I will let you know when I discoveref.

**Reply by barbudor** [reply]
If you are talking about this ESP32-C6 board : https://www.olimex.com/Products/IoT/ESP32-C6/ESP32-C6-EVB/open-source-hardware
don't buy : it is not supported by Tasmota yet

I don't see any other Olimex board with 4 relays

**Reply by barbudor** [reply]
Kincony seems to be a better option for now : https://templates.blakadder.com/kincony_KC868-A8.html

**Reply by Jason2866** [reply]
The Olimex board is now a good option. C6 is supported from Tasmota.

**Reply by jeroenst** [reply]
Or the dingtian board https://templates.blakadder.com/dingtian_DT-R008.html

Beware when ordering you have to ask the seller for an unlocked esp32 otherwise it's not possible to flash Tasmota unfortunately.

**Reply by Jason2866** [reply]
I received the Dingtian DT-R008 last week. Indeed nice one. Will post the compile env setup (in Dingtian thread) to have Ethernet with core 3.0.0 working.

**Comment by tk5ep** [comment]
Hi,
Received 2 of these beasts today.
I tried to flash with the tasmona32.bin file from here : https://github.com/arendst/Tasmota/releases/tag/v13.3.0

But it didn't succeed : Program flash size is larger than real flash size

I'm totally new in Tasmota, so forgive me if the question seems dumb.
Thanks,

**Reply by sfromis** [reply]
First wired flashing for ESP32 should be with `tasmota32.factory.bin` at offset 0, but recommended is to use the [Tasmota Web Installer](https://tasmota.github.io/install/) which makes the selection automatically. The binaries without `.factory` are meant for OTA upgrades, not for wired flashing. Note that many older flashing tools are not suitable for ESP32, but you told no details of what you did.

**Reply by tk5ep** [reply]
Thanks for fast answer.
You're right, some infos missing.
I've been able to use and test the modules. I've made some rules, etc.. So, they are working.

They came with Tasmota 12.3.1.2.
I tried an firmware upgrade from Tasmota and tasmota32.bin, that failed.

Then, I took the bin file I've found on the Rocketcontroller.com site, and that works.

BTW, is an upgrade really necessary and does it bring something ?

Thanks for any help.

**Reply by Jason2866** [reply]
**The first rule of upgrading: If it ain't broke, don't fix it!**

**Reply by sfromis** [reply]
If you want to upgrade, see https://github.com/arendst/Tasmota/discussions/18983

**Comment by jeroenst** [comment]
Still a nice module but unfortunately no CE or UL certification. So only use able for hobby projects.

**Comment by tk5ep** [comment]
Hi all,
I have one of these modules and it was working fine on my workbench.
This morning, I wanted to install it for real life. At first it was working and connected to the WiFi. OUTPUT LEDs lighted up as programmed.

Unfortunately, I had a bad +12V connection and had several shiort consecutive 12V power cuts.
Until that the module is like dead.

Now, only the PWR LEDs is ON. I do have +5V and +3.3V on the connector, and the RESET pushbutton seems to work as the current consumption drops to zero.

Any guess how I can bring the beast back to life ?
Thanks,

**Reply by Th0maz** [reply]
Multiple consecutive power cycles might have caused a settings reset to defaults. Can you connect to the device's wifi access point? tasmota-......
Then you can reconfigure it.

**Reply by tk5ep** [reply]
Hi , thanks for the answer.
No, unfortunately there WiFi does not work at all. (LED is OFF)

I will try to open the module and see if I can connect a serial port on the board.

**Comment by madduck** [comment]
Has anyone of you figured out if it's possible to disassociate the inputs from the relays? I want to be able to react to input events and drive outputs all from software only. Is this possible?

**Reply by Th0maz** [reply]
SetOption73
Detach buttons from relays and send multi-press and hold MQTT messages instead
0 = disable (default)
1 = enable
Example message: {"Button1":{"Action":"SINGLE"}}

**Reply by madduck** [reply]
> SetOption73 Detach buttons from relays and send multi-press and hold MQTT messages instead 0 = disable (default) 1 = enable Example message: {"Button1":{"Action":"SINGLE"}}

Thanks for your response. Unfortunately: I tried that, but it shows no effect. I have option 73 turned on, and switching the input still switches the output without any rules or anything I can discern.

**Reply by sfromis** [reply]
As the inputs are not "Button" but "Switch", it is `SetOption114` instead of `SetOption73`. To the extent you have rules to cover the switch states, this will automatically deactivate the default actions, also without `SetOption114`.
